components/org.wso2.carbon.identity.provider/src/main/java/org/wso2/carbon/identity/provider/AttributeCallbackHandler.java

```diff
@@ -147,8 +147,12 @@ public void handle(SAMLCallback callback) throws SAMLException {
 
                             if (StringUtils.isNotBlank(remoteClaimSuffixValue) &&
                                     StringUtils.isNotBlank(remoteClaimPrefixValue)) {
-
-                                if(!authenticatedUser.isFederatedUser()) {
+                                // WS trust flow does not set the authenticated user property.
+                                if (isHandlerCalledFromWSTrustSTSFlow(attrCallback)) {
+                                    localClaimValue = IdentityProviderServiceComponent.getRealmService().
+                                            getBootstrapRealm().getUserStoreManager().
+                                            getUserClaimValue(userIdentifier, localClaimUri, "default");
```

Comment:
This is the default profile right? Don't we have a constant for that?

Reply:
Fixed in 1d88701bc55af2dc361d6349b99e291e274c92da.